docs/rfc/rfc-1.adoc

```diff
@@ -8,21 +8,23 @@ toc::[]
 
 == Overview
 
-=== Minting
-
-Stakers periodically create 51-of-100 (subject to change) ecdsa-backed wallets
-to hold frozen BTC assets to maintain the peg. Depositors send BTC funds to the
-most-recently-created-wallet by using pay-to-script-hash (P2SH) which contains
-hashed information about the depositor's minting ethereum address. Once the
-block is mined, the depositor reveals their desired ethereum minting address to
-the ethereum chain. The TBTC system listens for these sorts of messages and
-when it gets one, it checks the bitcoin network to make sure the funds line up.
-If everything checks out, then we mint!
+=== Depositing
+
+Stakers periodically create <<group-size,51-of-100>> ecdsa-backed wallets
+to hold frozen BTC assets to maintain account balances. Depositors send BTC
+funds to the most-recently-created-wallet by using pay-to-script-hash (P2SH)
+which contains hashed information about the depositor's minting ethereum
+address. Once the block is mined, the depositor reveals their desired ethereum
+minting address to the ethereum chain. The bridge listens for these sorts
+of messages and when it gets one, it checks the bitcoin network to make sure
+the funds line up. If everything checks out, then we update your ethereum-side
+account balance.
 
 === Redemption
 
-A user supplies tBTCv2 and a bitcoin address. The system burns the tBTCv2 and
-releases the equivalent amount of bitcoin to the user from a user-supplied bitcoin wallet.
+A user with an account balance supplies a bitcoin address. The system decreases
+their account balance releases the equivalent amount of bitcoin to the user
+from a user-supplied bitcoin wallet.
```

Comment:
The wallet should probably be assigned by the bridge, not supplied by the user

Reply:
Can you elaborate more here?

Reply:
Instead of `releases the equivalent amount of bitcoin to the user from a user-supplied bitcoin wallet` the bridge should determine the correct wallet to release the BTC from, and release them _to_ the BTC address specified by the user.

Reply:
If you mean on-chain, I think that can get expensive since we need to find a wallet appropriate for the size of their redemption. Is there an attack that you have in mind? Otherwise it's simpler/cheaper to let the user/dApp specify the wallet, I think

Reply:
why wouldn't "the oldest wallet with enough BTC to cover the redemption" work? if we have a minimum BTC amount per wallet and close any wallet falling below that number, there shouldn't be too many small wallets to iterate through (and the iterations are storage reads, meaning that we'd be able to go through 25 wallets before the cost of the reads exceeds the base cost of a transaction)

it would also be possible to store the wallet information in a tree not unlike what we use in sortition pools, ensuring that no matter how many wallets we have active the cost of finding a wallet is always logarithmic in the maximum number of wallets over the system's lifetime

in the case of user-provided wallets, we could run into things like race conditions where two redemptions specify the same wallet but it doesn't have enough to cover both of them; in that case we'd either need to reject one redemption and force the user to try it again, or find a suitable wallet anyway

Reply:
I'm sold. @pdyraga @mhluongo ?

Reply:
Updated in https://github.com/keep-network/tbtc-v2/pull/45/commits/52beba8c7448d2161b6bfdfa071a85b1c3c274bb